astropy/visualization/tests/test_units.py

```diff
@@ -37,6 +37,20 @@ def test_units():
     plt.clf()
 
 
+@pytest.mark.skipif('not HAS_PLT')
+def test_units_axhline():
+    plt.figure()
+
+    with quantity_support():
+        plt.plot([1, 2, 3] * u.m, [3, 4, 5] * u.kg)
+        plt.axhline(10 * u.cm)
```

Comment:
A horizontal line would have to use the units of the vertical axis, no? So, maybe `10 * u.g`?

Reply:
Woops, that's right. I've tried that, and it throws a different error. I think there's going to need to be a bit more fixing on the Matplotlib side to use axhline actually. I'll let you know when I've had a chance to make it work properley!

Reply:
Okay, with https://github.com/matplotlib/matplotlib/pull/10143 and this PR `axvline` and `axhline` work properly with units. Shall I wait until the Matplotlib one is merged, and then come back to this PR?